src/org/lwjgl/opengl/awt/PlatformLinuxGLCanvas.java

```diff
@@ -48,7 +50,13 @@ private long create(int depth, GLData attribs, GLData effective) throws AWTExcep
 			// No framebuffer configurations supported!
 			throw new AWTException("No supported framebuffer configurations found");
 		}
-		long context = glXCreateNewContext(display, fbConfigs.get(0), GLX_RGBA_TYPE, NULL, true);
+		
+		long context;
+		if(attribs.shareContext != null){
+	    	context = glXCreateContext(display, glXGetVisualFromFBConfig(display, fbConfigs.get(0)), attribs.shareContext.context, true);
```

Comment:
Why did you switch the whole function being used? Have you had a look at the actual parameters of [glXCreateNewContext](https://www.khronos.org/registry/OpenGL-Refpages/gl2.1/xhtml/glXCreateNewContext.xml)? The fourth parameter is just the shared `GLXContext`.

Reply:
Oh i did not look into that, I was guessing from the name of the function that it was not capable of doing what i wanted (and the fact that it was not implemented yet even though being so trivial).

I'll fix that.